bundles/core/org.eclipse.smarthome.core.thing.test/src/test/groovy/org/eclipse/smarthome/core/thing/firmware/ProgressCallbackTest.groovy

```diff
@@ -0,0 +1,79 @@
+/**
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.thing.firmware
+
+import static org.hamcrest.CoreMatchers.*
+import static org.junit.Assert.*
+
+import org.eclipse.smarthome.core.events.EventPublisher;
+import org.eclipse.smarthome.core.thing.ThingTypeUID
+import org.eclipse.smarthome.core.thing.ThingUID
+import org.eclipse.smarthome.core.thing.binding.firmware.FirmwareUID
+import org.eclipse.smarthome.core.thing.binding.firmware.ProgressCallback;
+import org.eclipse.smarthome.core.thing.binding.firmware.ProgressStep;
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+/**
+ * Testing the {@link ProgressCallback}.
+ *
+ * @author Christoph Knauf - Initial contribution
+ */
+public final class ProgressCallbackTest {
+
+    ProgressCallback sut
+    boolean eventPosted = false 
+    
+    @Before
+    void setUp(){
+        def typeUID = new ThingTypeUID("thing:type")
+        def publisher = [
+            post : { event ->
+                eventPosted = true
+            }
+        ] as EventPublisher
+        sut = new ProgressCallbackImpl(null, publisher, null, new ThingUID(typeUID, "thingid"), new FirmwareUID(typeUID, "1"), null)
+    }
+    
+    @Test
+    void 'assert that setting the progess to pending results in a FirmwareUpdateProgressInfo event'(){
+        sut.defineSequence(ProgressStep.DOWNLOADING, ProgressStep.TRANSFERRING)
+        sut.pending()
+        assertThat eventPosted, is(true)
```

Comment:
I would also expect that the content of the sent event is checked (e.g. what is the current step, is pending true or false)